compare module done

Ansarinamrah opened the pull request.

Ansarinamrah committed:
compare module done

skhasan071 commented on lib/features/compare/presentation/compare_school_view.dart:
```diff
@@ -1,155 +1,110 @@
+      appBar: SAppBar(
+        leading: SIcon(
+          icon: Icons.keyboard_arrow_left,
+          onTap: () => Navigator.of(context).pop(),
```
Instead Navigator directly pop context using context.pop

skhasan071 commented on lib/features/compare/presentation/compare_school_view.dart:
```diff
@@ -1,155 +1,110 @@
   @override
   Widget build(BuildContext context) {
+    final extra = GoRouterState.of(context).extra as Map<String, dynamic>?;
+    final Map<String, dynamic>? s1 = extra?['school1'] as Map<String, dynamic>?;
```
Why Map<String, dynamic>? isn't the parameter are similar to school model, use SchoolModel instead. if its matches SchoolCardModel then use that.

skhasan071 commented on lib/core/navigation/router.dart:
```diff
@@ -224,24 +224,18 @@ class AppRouter {
-  },
-),
+ GoRoute(
+      path: '/compare-with',
```
Pass school ids over here

skhasan071 commented on lib/features/compare/presentation/compare_school_view.dart:
```diff
@@ -1,155 +1,110 @@
 
   @override
   Widget build(BuildContext context) {
+    final extra = GoRouterState.of(context).extra as Map<String, dynamic>?;
```
Make the API call in the compare page.

skhasan071 commented on lib/features/compare/data/data_source/data_source.dart:
```diff
@@ -0,0 +1,10 @@
+import 'package:tc_sa/core/network/typedef.dart';
+
+abstract class CompareDataSource {
+  ResultFuture<Map<String, dynamic>> compareSchools({
```
Use Model instead of MAP

skhasan071 commented on lib/features/compare/presentation/compare_with_view.dart:
```diff
@@ -1,289 +1,218 @@
+              ),
+              
+              // Loading overlay
+              if (_isComparing)
```
No extra flags needed is loading will work

skhasan071 commented on lib/features/compare/presentation/view_models/compare_view_model.dart:
```diff
@@ -0,0 +1,57 @@
+class CompareViewModel extends ViewStateProvider {
+  final CompareDataSourceImpl _dataSource;
+
+  CompareViewModel({CompareDataSourceImpl? dataSource})
```
U can't pass anything to view model constructor

skhasan071 commented on lib/features/compare/presentation/view_models/compare_view_model.dart:
```diff
@@ -0,0 +1,57 @@
+    ) {
+      _school1 = res['school1'] as Map<String, dynamic>?;
+      _school2 = res['school2'] as Map<String, dynamic>?;
+      notifyListeners();
```
Dont notify the listeners over here as changing the state will automatically notify

skhasan071 commented on lib/features/compare/presentation/widgets/compareWith_widgets.dart:
```diff
@@ -0,0 +1,195 @@
+                const SizedBox(width: 4),
+                Text(
+                  showShortlistedOnly ? "Showing Shortlisted" : "Shortlisted Schools",
+                  style: TextStyle(
```
STextStyles instead of TextStyles

skhasan071 commented on lib/features/compare/presentation/widgets/compareWith_widgets.dart:
```diff
@@ -0,0 +1,195 @@
+          Expanded(
+            child: Text(
+              "Base School: $baseName",
+              style: const TextStyle(fontWeight: FontWeight.w600),
```
STextStyles

skhasan071 commented on lib/features/compare/presentation/widgets/compare_widgets.dart:
```diff
@@ -1,246 +1,113 @@
-          "$percent%",
-          style: const TextStyle(color: Colors.black, fontWeight: FontWeight.bold),
+          'Compare. Decide. Succeed!',
+          style: TextStyle(fontWeight: FontWeight.bold, fontSize: 17),
```
STextStlyes

skhasan071 commented on lib/features/detailPages/overview/presentation/overview_view.dart:
```diff
@@ -197,20 +197,17 @@ class _SchoolDetailViewState extends State<SchoolDetailView>
                             Expanded(
```
use SButton

skhasan536 committed:
feat: Enhance school comparison functionality

This commit introduces `SchoolCompareModel` and its generated files for structured school comparison data. The comparison flow is updated to fetch data using school IDs and display it in `CompareSchools` view.

Key changes:
- Added `SchoolCompareModel` with `freezed` and `json_serializable` for data handling.
- Modified `CompareDataSource` and `CompareDataSourceImpl` to return `Map<String, SchoolCompareModel>`.
- Updated `CompareViewModel` to handle `SchoolCompareModel` and removed the `clear` method.
- Refactored `CompareSchools` to be a `StatefulWidget`, fetch comparison data in `initState`, and display it using `CompareViewModel`.
- Updated `CompareWith` view to pass school IDs to `CompareSchools` via route extras.
- Adjusted `CompareWithWidgets` to handle `Future<void> Function(String id)` for the `onCompare` callback.
- Modified `OverviewView` to pass school ID and name as a map in route extras.
- Updated `router.dart` to handle new extra types for compare routes.
- Minor UI adjustments in `compare_widgets.dart` for better presentation.
- Removed unused `device_preview` and related dependencies.

skhasan536 committed:
Refactor: Update navigation and layout in OverviewView

This commit updates the navigation method in `OverviewView` from `Navigator.of(context).pop()` to the more concise `context.pop()`.

Additionally, the layout of `InfoChip` widgets within the `OverviewView` has been adjusted. They are now wrapped in `Expanded` widgets to ensure they distribute available space evenly within their parent `Row`. The spacing between these chips has also been reduced from 16 to 8.

A commented-out line at the end of `router.dart` has been removed.
